### PR TITLE
Update public IP creation/reconcile

### LINKED ISSUE #122: GetIP creates extraneous public IP
/kind bug

**What steps did you take and what happened:**
`GetIP` in the `deployer` package utilizes the `CreateOrUpdatePublicIPAddress` method, which results in creating an additional (unused) Standard Public IP.

We should create a `GetPublicIPAddress` method to retrieve the public IP FQDN instead.

/assign
/milestone mvp
/priority critical-urgent

### PR DESCRIPTION
**Pull request by justaugustus**
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

**What this PR does / why we need it**:
- deployer: Improve FQDN retrieval in GetIP/GetKubeConfig
- network: Update GetPublicIPName usage
- network: Update public IP reconcile checks

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #122 
Partially addresses #115 

**Release note**:

```release-note
Update public IP creation/reconcile
```

/milestone mvp
/priority critical-urgent
/hold

### CODE REVIEW

File: pkg/cloud/azure/services/network/publicipaddress.go [hunk 1]
```diff
@@ -25,7 +25,23 @@ import (
...
+// GetPublicIPAddress retrieves the Public IP address resource.
+func (s *Service) GetPublicIPAddress(resourceGroup, IPName string) (network.PublicIPAddress, error) {
+	pip, err := s.scope.PublicIPAddresses.Get(
+		s.scope.Context,
+		resourceGroup,
+		IPName,
+		"",
+	)
+
+	if err != nil {
```
Comment (tariq1890) [inline pkg/cloud/azure/services/network/publicipaddress.go]: lines 37-42 are redundant. We can just do `return pip, nil` instead.

**Comment by tariq1890** [comment]
Just a minor nit. I will review more once the WIP flag is removed.

File: pkg/cloud/azure/services/network/publicipaddress.go (hunk 1 above)
Comment (tariq1890) [reply pkg/cloud/azure/services/network/publicipaddress.go]: or even return `scope.PublicIPAddresses.Get(s.scope.Context, resourceGroup, IPName, "")` .

Comment (justaugustus) [reply pkg/cloud/azure/services/network/publicipaddress.go]: Thanks for catching. I've wrapped the error (which is what I meant to do the first time around. :sweat_smile:)

**Comment by justaugustus** [comment]
Ready for review!
/assign @tariq1890 @soggiest 
/hold cancel

**Pull request merged**